### ISSUE TITLE
"OSError: CUDA_HOME environment variable is not set" when installing submodules/depth-diff-gaussian-rasterization

### ISSUE THREAD

**Issue by EarltShirt** [hustvl/4DGaussians#107]
I'm trying to install all the requirements, but both of the submodules show the same error message when trying to install the dependencies using the given pip command in the readme file.
When trying to execute `pip install -e submodules/depth-diff-gaussian-rasterization`,  the following error message appears:

```
Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error

  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [14 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "Path\to\4DGaussians\submodules\depth-diff-gaussian-rasterization\setup.py", line 21, in <module>
          CUDAExtension(
        File "Path\to\anaconda3\Lib\site-packages\torch\utils\cpp_extension.py", line 1074, in CUDAExtension 
          library_dirs += library_paths(cuda=True)
                          ^^^^^^^^^^^^^^^^^^^^^^^^
        File "Path\to\anaconda3\Lib\site-packages\torch\utils\cpp_extension.py", line 1208, in library_paths 
          paths.append(_join_cuda_home(lib_dir))
                       ^^^^^^^^^^^^^^^^^^^^^^^^
        File "Path\to\anaconda3\Lib\site-packages\torch\utils\cpp_extension.py", line 2407, in _join_cuda_home
          raise OSError('CUDA_HOME environment variable is not set. '
      OSError: CUDA_HOME environment variable is not set. Please set it to your CUDA install root.
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
```

even though is set it through the windows environment variable window as well as the `set CUDA_HOME="..."` command. It is also found by my system in the command prompt, powershell and in my conda environment. When executing `echo %CUDA_HOME%`, the right path appears. It just can't be found when launching the setup.py script.

I'm working on Windows 11 using conda, I tried it using different python versions and many different torch versions. 

Thanks for your time.

**Comment by guanjunwu** [comment]
Hi, 
Do you install `torch` with the correct versions?

**Comment by EarltShirt** [comment]
And when using conda in order to install torch (as recommended on pytorch website) using `conda install pytorch==1.13.1 torchvision==0.14.1 torchaudio==0.13.1 pytorch-cuda=11.6 -c pytorch -c nvidia` I get another error during the build of diff_gaussian_rasterization extension :

```
  Preparing metadata (setup.py) ... done
Installing collected packages: diff-gaussian-rasterization
  Running setup.py develop for diff-gaussian-rasterization
    error: subprocess-exited-with-error
    
    × python setup.py develop did not run successfully.
    │ exit code: 1
    ╰─> [96 lines of output]
        running develop
        running egg_info
        writing diff_gaussian_rasterization.egg-info\PKG-INFO
        writing dependency_links to diff_gaussian_rasterization.egg-info\dependency_links.txt
        writing top-level names to diff_gaussian_rasterization.egg-info\top_level.txt
        reading manifest file 'diff_gaussian_rasterization.egg-info\SOURCES.txt'
        adding license file 'LICENSE.md'
        writing manifest file 'diff_gaussian_rasterization.egg-info\SOURCES.txt'
        running build_ext
        building 'diff_gaussian_rasterization._C' extension
        Path\to\anaconda3\envs\Gaussians4D\lib\site-packages\setuptools\command\easy_install.py:147: EasyInstallDeprecationWarning: easy_install command is deprecated. Use build and pip and other standards-based tools.
          EasyInstallDeprecationWarning,
        Path\to\anaconda3\envs\Gaussians4D\lib\site-packages\setuptools\command\install.py:37: SetuptoolsDeprecationWarning: setup.py install is deprecated. Use build and pip and other standards-based tools.
          setuptools.SetuptoolsDeprecationWarning,
        Path\to\anaconda3\envs\Gaussians4D\lib\site-packages\torch\utils\cpp_extension.py:358: UserWarning: Error checking compiler version for cl: [WinError 2] Le fichier spécifié est introuvable
          warnings.warn(f'Error checking compiler version for {compiler}: {error}')
        Emitting ninja build file Path\to\repos\4DGaussians\submodules\depth-diff-gaussian-rasterization\build\temp.win-amd64-cpython-37\Release\build.ninja...
        Compiling objects...
        Allowing ninja to set a default number of workers... (overridable by setting the environment variable MAX_JOBS=N)
        [1/1] C:\Program Files\NVIDIA GPU Computing Toolkit\CUDA\v11.6\bin\nvcc --generate-dependencies-with-compile --dependency-output 
        
        [......]
```

**Comment by EarltShirt** [comment]
had the wrong graphics card, everything is working now as i upgraded to a RTX 3060

**Comment by TonyDua** [comment]
If someone encounters the same problem, check if you have installed PyTorch incorrectly. I made a mistake when looking for previous versions at https://pytorch.org/get-started/previous-versions/ and habitually copied the pip command without paying attention, which resulted in copying the Wheel OSX command. That was silly of me. I uninstalled the OSX package and re-ran the command under "Linux and Windows," which solved the issue.
